### PR TITLE
[Vanity URLs] - SSL verify, region update message, debug logging

### PR DESCRIPTION
**Pull request by AndrewAsseily**
*Description of changes:*
- Plumb SSL verify/CA bundle settings through the resolver
- Print "SSO region updated to <region>" when login updates `sso_region`
- Add debug logging for HEAD/GET requests during URL resolution

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: awscli/customizations/configure/sso_commands.py [hunk 1]
```diff
@@ -318,7 +318,9 @@ def _run_main(self, parsed_args, parsed_globals):
         on_pending_authorization = None
         if parsed_args.no_browser:
             on_pending_authorization = PrintOnlyHandler()
-        sso_registration_args = self._prompt_for_sso_registration_args()
+        sso_registration_args = self._prompt_for_sso_registration_args(
+            verify=parsed_globals.verify_ssl,
```
Comment (ashovlin) [inline awscli/customizations/configure/sso_commands.py]: I don't think this will pick up `ca_bundle` in the config file or `AWS_CA_BUNDLE` in the environment.

If not, maybe we can reuse `session.get_config_variable('ca_bundle')` since I think that'd check the full hierarchy.

File: tests/unit/customizations/sso/test_resolve.py [hunk 2]
```diff
@@ -272,7 +274,9 @@ def test_custom_timeout_passed_to_follow_redirect(self):
                 timeout=5,
             )
             mock_follow.assert_called_once_with(
-                'https://aws.mycompany.com', timeout=5
+                'https://aws.mycompany.com',
+                timeout=5,
+                verify=None,
```
Comment (ashovlin) [inline tests/unit/customizations/sso/test_resolve.py:280]: Especially tied to the other comment, can you add a case or two that tests a non-None `verify`?

**Review by ashovlin — CHANGES_REQUESTED** [review]

File: awscli/customizations/configure/sso_commands.py (hunk 1 above)
Comment (AndrewAsseily) [reply awscli/customizations/configure/sso_commands.py]: Good catch thanks!

**Review by ashovlin — APPROVED** [review]
